zeebe/scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorTask.java

```diff
@@ -487,11 +509,38 @@ public CompletableActorFuture<Void> getStartingFuture() {
 
   public void addSubscription(final ActorSubscription subscription) {
     ensureCalledFromActorThread("addSubscription(ActorSubscription)");
+    LOG.info(
+        "FINDME: adding subscription {} from actor {} : thread {}, {}",
+        subscription,
+        actor.getName(),
+        ActorThread.current(),
+        Thread.currentThread().threadId());
+
+    if (actor.threadId == 0 || actor.threadId != Thread.currentThread().threadId()) {
```

Comment:
🔧 We need to assign actor.threadId to a stable value, from the logs it appears that it's different from 0 in the if condition and then becomes 0 when we log it.